react/I18n/index.jsx

```diff
@@ -74,14 +74,20 @@ I18n.childContextTypes = i18nContextTypes
 
 // higher order decorator for components that need `t` and/or `f`
 export const translate = () => WrappedComponent => {
-  const Wrapper = (props, context) => (
-    <WrappedComponent
-      {...props}
-      t={context.t}
-      f={context.f}
-      lang={context.lang}
-    />
-  )
+  const Wrapper = (props, context) => {
+    return (
+      <WrappedComponent
+        {...props}
+        t={context.t}
+        f={context.f}
+        lang={context.lang}
+      />
+    )
+  }
+  Wrapper.propTypes = {
+    ...(WrappedComponent.propTypes || {}),
```

Comment:
That sounds weird but you can actually destructure `undefined`:

```js
> { ...undefined }
{}
```

So you can just do

```js
...WrappedComponent.propTypes
```

even if `WrappedComponent.propTypes` is undefined.

Reply:
Fixed by b39adba1c709803f39830c506dd0552d71b334f3